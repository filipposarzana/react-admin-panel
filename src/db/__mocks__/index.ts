import { User } from '~/db'

export const users: User[] = [
  {
    friends: [
      {
        name: 'A Friend',
        uuid: 'acc82963-61c3-37ec-b334-e8f6bf6b7f57',
      },
    ],
    name: 'A User',
    uuid: '6bb0295f-8665-30a1-af76-6d1236b88971',
  },
  {
    friends: [],
    name: 'A User Friend',
    uuid: 'acc82963-61c3-37ec-b334-e8f6bf6b7f57',
  },
]
