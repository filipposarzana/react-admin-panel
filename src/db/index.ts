export type BaseUser = {
  name: string
  uuid: string
}

export type User = BaseUser & {
  friends: BaseUser[]
}

export type UpdatableUser = Omit<User, 'uuid'>

export const users: User[] = []
