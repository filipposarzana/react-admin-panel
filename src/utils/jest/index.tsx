import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
