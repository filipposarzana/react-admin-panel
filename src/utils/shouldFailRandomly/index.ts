export const shouldFailRandomly = () => Math.round(Math.random()) === 0
