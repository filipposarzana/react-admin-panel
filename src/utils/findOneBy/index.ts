import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
