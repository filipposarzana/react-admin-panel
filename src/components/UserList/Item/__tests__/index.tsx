import { users } from '~/db'
import { renderer } from '~/utils/jest'
import { noop } from '~/utils/noop'
import { UserListItem, Props } from '..'

const fixtures: [string, Props][] = [
  ['matches snapshot without remove Button', { index: 0, onRemove: noop, user: users[0] }],
  ['matches snapshot without remove Button', { index: 0, onRemove: jest.fn(), user: users[0] }],
]

jest.mock('~/db')

afterAll(jest.clearAllMocks)

describe('UserListItem', () => {
  fixtures.forEach(([title, props]) => {
    it(title, () => {
      const { output } = renderer.shallow(<UserListItem {...props} />)

      expect(output).toMatchSnapshot()
    })
  })
})
