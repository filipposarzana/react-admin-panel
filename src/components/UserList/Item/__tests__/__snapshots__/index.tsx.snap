// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListItem matches snapshot without remove Button 1`] = `
<SectionBodyListItem
  index={0}
>
  <styled.div
    align="center"
    direction="row"
    justify="space-between"
  >
    <Link
      href="users/edit/6bb0295f-8665-30a1-af76-6d1236b88971"
    >
      <styled.span
        as="a"
      >
        <UserListItemName
          name="A User"
        />
        <br />
        <UserListItemUuid
          uuid="6bb0295f-8665-30a1-af76-6d1236b88971"
        />
      </styled.span>
    </Link>
  </styled.div>
</SectionBodyListItem>
`;

exports[`UserListItem matches snapshot without remove Button 2`] = `
<SectionBodyListItem
  index={0}
>
  <styled.div
    align="center"
    direction="row"
    justify="space-between"
  >
    <Link
      href="users/edit/6bb0295f-8665-30a1-af76-6d1236b88971"
    >
      <styled.span
        as="a"
      >
        <UserListItemName
          name="A User"
        />
        <br />
        <UserListItemUuid
          uuid="6bb0295f-8665-30a1-af76-6d1236b88971"
        />
      </styled.span>
    </Link>
    <styled.div
      pl={8}
    >
      <UserListItemRemoveButton
        onRemove={[MockFunction]}
        uuid="6bb0295f-8665-30a1-af76-6d1236b88971"
      />
    </styled.div>
  </styled.div>
</SectionBodyListItem>
`;
